WcaOnRails/spec/features/manage_competition_spec.rb

```diff
@@ -0,0 +1,60 @@
+# frozen_string_literal: true
+require "rails_helper"
+
+RSpec.feature "Manage competition" do
+  let(:delegate) { FactoryGirl.create(:delegate) }
+  let(:cloned_delegate) { FactoryGirl.create(:delegate) }
+  let(:competition_to_clone) { FactoryGirl.create :competition, cityName: 'Melbourne', delegates: [cloned_delegate], showAtAll: true }
+
+  before :each do
+    sign_in delegate
+  end
+
+  context "create" do
+    it 'can create competition', js: true do
+      visit "/competitions/new"
+
+      fill_in "Name", with: "New Comp 2015"
+
+      click_button "Create Competition"
+
+      expect(Competition.all.length).to eq 1
```

Comment:
That's a race condition waiting to happen. I'll explain why:

With JavaScript tests the actual web server runs in a separate process as this RSpec test. So when clicking on the "Create Competition" button the web server needs to be fast enough and insert the record in the database before RSpec executes this line.
You can solve this by first asserting some things about the current page (e.g. `expect(page).to have_content "Competition successfully created"`) which forces the RSpec process to wait.